rebasehelper/specfile.py

```diff
@@ -195,23 +195,28 @@ def guess_category():
     # TAG HELPER METHODS #
     ######################
 
-    def tag(self, name: str, section: Optional[str] = None) -> Optional[Tag]:
+    def tag(self, name: str, section: Optional[Union[str, int]] = None) -> Optional[Tag]:
         """Returns the first non-unique tag."""
-        tags = self.tags.filter(section=section, name=name)
+        if isinstance(section, str):
+            tags = self.tags.filter(section=section, name=name)
+        else:
+            tags = self.tags.filter(section_index=section, name=name)
         return next(tags, None)
 
-    def get_raw_tag_value(self, tag_name: str, section: Optional[str] = None) -> Optional[str]:
+    def get_raw_tag_value(self, tag_name: str, section: Optional[Union[str, int]] = None) -> Optional[str]:
         tag = self.tag(tag_name, section)
         if not tag:
             return None
-        return self.spec_content.section(tag.section)[tag.line][slice(*tag.value_span)]
+        section = self.spec_content.section(section) if isinstance(section, str) else self.spec_content[section]
```

Comment:
This is not so great, perhaps `SpecContent` could do the same and accept both types.

Reply:
I thought about this one. `SpecContent` could accept string or int and either call `SpecContent.section()` or just return by index. What would then be the point of `SpecContent.section()`? Another thing to note is that `__getitem__` could then return None which would need to be handled when using it.

Reply:
`__getitem__` shoudn't return `None`, it shoud raise `KeyError` (or `IndexError`). There is no point in preserving the `section()` method, unless removing it would break the API. But I think we should strive to define a clean and consistent API, even at the cost of occasional incompatibility.

Reply:
Looking at this now, the line didn't make sense at all. The different behaviour of string and int in `section` is already handled in `SpecFile.tag()`. This line should then just use the obtained tag to determine the correct section. So it seems like there is no need to change `SpecContent`. Btw removing `section()` would break the API. IIRC packit uses the method somewhere.